engine/src/main/java/org/terasology/particles/ParticleSystemManagerImpl.java

```diff
@@ -0,0 +1,143 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.particles;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.Component;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeDeactivateComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnActivatedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.module.sandbox.API;
+import org.terasology.particles.components.ParticleEmitterComponent;
+import org.terasology.particles.components.ParticleSystemComponent;
+import org.terasology.particles.events.ParticleSystemUpdateEvent;
+import org.terasology.particles.functions.affectors.AffectorFunction;
+import org.terasology.particles.functions.generators.GeneratorFunction;
+import org.terasology.particles.rendering.ParticleRenderer;
+import org.terasology.particles.updating.ParticleUpdater;
+import org.terasology.physics.Physics;
+import org.terasology.registry.In;
+import org.terasology.registry.Share;
+import org.terasology.world.WorldProvider;
+
+import java.util.stream.Stream;
+
+/**
+ * See ParticleSystemManager for more information.
+ */
+@API
+@Share(ParticleSystemManager.class)
+@RegisterSystem(RegisterMode.CLIENT)
+public class ParticleSystemManagerImpl extends BaseComponentSystem implements UpdateSubscriberSystem, ParticleSystemManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(ParticleSystemManagerImpl.class);
+
+    @In
+    private WorldProvider worldProvider;
+
+    @In
+    private EntityManager entityManager;
+
+    @In
+    private Physics physics;
+
+    private ParticleUpdater particleUpdater;
+
+    private BiMap<Class<Component>, GeneratorFunction> registeredGeneratorFunctions = HashBiMap.create();
+    private BiMap<Class<Component>, AffectorFunction> registeredAffectorFunctions = HashBiMap.create();
+
+
+    @ReceiveEvent(components = {ParticleSystemComponent.class})
+    public void onSystemActivated(OnActivatedComponent event, EntityRef entity, ParticleSystemComponent particleSystemComponent) {
```

Comment:
Yep with the `Component` in the method header it is assumed to also count for `ReceiveEvent` :-)

Confusing class names for existing stuff is documented in #2443. `OnActivatedComponent ` is pre-existing - I just added it to the list of things to cover in that other issue. It is an API change for sure so needs to wait till engine v2.0.0